blockchain/process.go

```diff
@@ -42,7 +42,8 @@ const (
 func (b *BlockChain) blockExists(hash *chainhash.Hash) (bool, error) {
 	// Check block index first (could be main chain or side chain blocks).
 	if b.index.HaveBlock(hash) {
-		return true, nil
+		node := b.index.LookupNode(hash)
+		return true && node.status.HaveData(), nil
 	}
 
 	// Check in the database.
```

Comment:
@kcalvinalvin in the blockExists function, do I need to check whether block data exists for the block stored in database or not. Can a node be stored in the database, just with its header information?